### ISSUE TITLE
Searchbox: Can't click suggestions with Narrator Touch

### ISSUE THREAD

**Issue by rigdern** [winjs/winjs#543]
### Impact

You cannot select Searchbox suggestions with Narrator Touch. This bug has existed since WinJS 2.0.
### Repro Steps
1. Go to http://try.buildwinjs.com/#searchbox
2. Put focus into the search box
   1. Hover over the searchbox with your finger until there's a blue rectangle around it
   2. Double tap to put focus into the searchbox
3. Type a letter to bring up the suggestions
4. Select a suggestion with Narrator
   1. Hover over "Mint" with your finger until there's a blue rectangle around it
   2. Double tap to select "Mint"

**Expected**: "Mint" is written in the Searchbox and the suggestion flyout goes away (same behavior as though you clicked "Mint" with a mouse)

**Actual**: Nothing happens
### Details

Here are some things I noticed that might be useful to know when fixing this:
- [preventDefault in _flyoutPointerDownHandler](https://github.com/winjs/winjs/blob/8334c4a2e0fcfd985f82d1d2c376e89e48a864fc/src/js/WinJS/Controls/SearchBox.js#L1300)
  - It's goal is to prevent focus from leaving the `_inputElement`. However, this doesn't work when Narrator is used to invoke a suggestion because Narrator doesn't go thru pointer down.
  - This preventDefault has the possibly unintended side effect of preventing the `click` event from firing in WebKit
- The following two event handlers use `event.relatedTarget` but this property isn't supported in all browsers. It may cause dismissals at unexpected times. We should stop using it:
  - [_searchBoxFocusHandler](https://github.com/winjs/winjs/blob/8334c4a2e0fcfd985f82d1d2c376e89e48a864fc/src/js/WinJS/Controls/SearchBox.js#L1167)
  - [_searchBoxBlurHandler](https://github.com/winjs/winjs/blob/8334c4a2e0fcfd985f82d1d2c376e89e48a864fc/src/js/WinJS/Controls/SearchBox.js#L1190)
